cpp/src/plasma/client.h

```diff
@@ -95,6 +95,17 @@ class ARROW_EXPORT PlasmaClient {
   Status Create(const ObjectID& object_id, int64_t data_size, const uint8_t* metadata,
                 int64_t metadata_size, std::shared_ptr<Buffer>* data, int device_num = 0);
 
+  /// Create and seal an object in the object store. This is an optimization
+  /// which allows small objects to be created quickly with fewer messages to
+  /// the store.
+  ///
+  /// \param object_id The ID of the object to create.
+  /// \param data The data for the object to create.
+  /// \param metadata The metadata for the object to create.
+  /// \return The return status.
+  Status CreateAndSeal(const ObjectID& object_id, const std::string& data,
```

Comment:
question: why `string` data and not an arbitrary byte array with size?

Reply:
Strings are fine for small objects I think (they might incur an additional copy in some cases, i.e. if the user needs to create a new string to pass the data in)